src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -74,21 +72,47 @@ public DatasetController(ZipService zipService,
   })
   @PostMapping(value = "dataset/{name}/process", produces = APPLICATION_JSON_VALUE)
   @ResponseStatus(HttpStatus.ACCEPTED)
-  public DatasetIdDto processDataset(
+  public DatasetIdDto harvestDatasetFromFile(
       @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
       @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
       @ApiParam(value = "language of the dataset", required = true, defaultValue = "nl") @RequestParam Language language,
-      @ApiParam(value = "dataset records in a zip file", required = true) @RequestParam MultipartFile dataset) {
+      @ApiParam(value = "dataset records uploaded in a zip file", required = true) @RequestParam MultipartFile dataset) {
     checkArgument(namePattern.matcher(datasetName).matches(),
         "dataset name can only include letters, numbers, _ or - characters");
-    var records = zipService.parse(dataset);
+
+    List<ByteArrayInputStream> records = harvestService.harvest(dataset);
+
     checkArgument(records.size() < maxRecords,
         "Amount of records can not be more than " + maxRecords);
 
     var datasetObject = datasetService.createDataset(datasetName, country, language, records);
     return new DatasetIdDto(datasetObject);
   }
 
+  @ApiOperation("Process the given dataset")
+  @ApiResponses({
+          @ApiResponse(code = 202, message = MESSAGE_FOR_PROCESS_DATASET, response = Object.class)
+  })
+  @PostMapping(value = "dataset/{name}/process", produces = APPLICATION_JSON_VALUE)
+  @ResponseStatus(HttpStatus.ACCEPTED)
+  public void harvestDatasetFromURL(
+          @ApiParam(value = "name of the dataset", required = true) @PathVariable(value = "name") String datasetName,
+          @ApiParam(value = "country of the dataset", required = true, defaultValue = "Netherlands") @RequestParam Country country,
+          @ApiParam(value = "language of the dataset", required = true, defaultValue = "nl") @RequestParam Language language,
+          @ApiParam(value = "dataset records URL to download in a zip file", required = true) @RequestParam String url) {
+    checkArgument(namePattern.matcher(datasetName).matches(),
+            "dataset name can only include letters, numbers, _ or - characters");
+    try {
+      HttpRecordIterator iterator = harvestService.harvest(url);
```

Comment:
Your doubts about the `HttpRecordIterator` can be answered:
First of the iterator here is not an issue and the method `forEach` could be used to create a result of records that can be used down the line, similarly to the `harvestDatasetFromFile` method.
Secondly if you check the actual `HttpHarvesterImpl`, on the line `eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java:64` the same `HttpRecordIterator` is used in a similar fashion. So perhaps I think it's possible to even use the other harvestRecords method inside the `HarvestServiceImpl` by passing an `InputStream` if you'd like.